secure/storage/src/storage.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{error::Error, permissions::Permissions, value::Value};
+
+pub trait Storage {
+    /// Creates a new value if it does not exist fails only if there is some other issue.
+    fn create_if_not_exists(
+        &mut self,
+        key: &str,
+        value: Value,
+        permissions: &Permissions,
+    ) -> Result<(), Error> {
+        self.create(key, value, permissions).or_else(|err| {
+            if let Error::KeyAlreadyExists(_) = err {
+                Ok(())
+            } else {
+                Err(err)
+            }
+        })
+    }
+    /// Creates a new value in storage and fails if it already exists
+    fn create(&mut self, key: &str, value: Value, permissions: &Permissions) -> Result<(), Error>;
+    /// Retreives a value from storage and fails if invalid permiossions or it does not exist
+    fn get(&self, key: &str) -> Result<Value, Error>;
```

Comment:
why `get` and `set` don't take a `Permission`?

Reply:
I understand this with offline discussion, it's probably worth an examples - using public key as Id(String) and each process would hold corresponding private keys for get/set.

create API is used to set what PublicKey could access the data, and get/set would use the private key internally for access.